app/scripts/controllers/edit/buildConfig.js

```diff
@@ -117,6 +118,75 @@ angular.module('openshiftConsole')
       "SerialLatestOnly"
     ];
 
+    $scope.buildHookTypes = [{
+      id: "command",
+      label: "Command"
+    }, {
+      id: "script",
+      label: "Shell Script"
+    }, {
+      id: "args",
+      label: "Arguments to default image entry point"
+    }, {
+      id: "commandArgs",
+      label: "Command with arguments"
+    }, {
+      id: "scriptArgs",
+      label: "Shell script with arguments"
+    }];
+
+    $scope.buildHookSelection = {
+      type: {}
+    };
+
+    var getInitialBuildHookSelection = function() {
+      var hasArgs = !_.isEmpty(_.get($scope, 'buildConfig.spec.postCommit.args'));
+      var hasCommand = !_.isEmpty(_.get($scope, 'buildConfig.spec.postCommit.command'));
+      var hasScript = !!_.get($scope, 'buildConfig.spec.postCommit.script');
+      $scope.view.hasHooks = hasArgs || hasCommand || hasScript;
+
+      if (hasArgs && hasCommand) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'commandArgs' });
+      } else if (hasArgs && hasScript) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'scriptArgs' });
+      } else if (hasArgs) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'args' });
+      } else if (hasScript) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'script' });
+      } else {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'command' });
+      }
+    };
+
+    var clearIncompatibleBuildHookFields = function() {
+      if ($scope.view.hasHooks) {
+        switch ($scope.buildHookSelection.type.id) {
+          case "script":
+            delete $scope.updatedBuildConfig.spec.postCommit.command;
+            delete $scope.updatedBuildConfig.spec.postCommit.args;
+            break;
+          case "command":
+            delete $scope.updatedBuildConfig.spec.postCommit.script;
+            delete $scope.updatedBuildConfig.spec.postCommit.args;
+            break;
+          case "args":
+            delete $scope.updatedBuildConfig.spec.postCommit.script;
+            delete $scope.updatedBuildConfig.spec.postCommit.command;
+            break;
+          case "scriptArgs":
+            delete $scope.updatedBuildConfig.spec.postCommit.command;
+            break;
+          case "commandArgs":
+            delete $scope.updatedBuildConfig.spec.postCommit.script;
+            break;
+        }
+      } else {
+        $scope.updatedBuildConfig.spec.postCommit.command = [];
+        $scope.updatedBuildConfig.spec.postCommit.args = [];
+        $scope.updatedBuildConfig.spec.postCommit.script = "";
+      }
```

Comment:
What you have here is fine, although I might handle the simple case first and return early. It makes the code a little easier to understand because it avoids a big `if` block.

```js
var clearIncompatibleBuildHookFields = function() {
  if (!$scope.view.hasHooks) {
    $scope.updatedBuildConfig.spec.postCommit.command = [];
    $scope.updatedBuildConfig.spec.postCommit.args = [];
    $scope.updatedBuildConfig.spec.postCommit.script = "";
    return;
  }
  
  switch ($scope.buildHookSelection.type.id) {
    // ...
  }
};
```

Probably should delete the keys for `!hasHooks` case as well instead of setting them to empty values.